pkg/hwinfo/hwinfo_nonwindows.go

```diff
@@ -63,9 +64,18 @@ func retrieveInfoUsingDmiDecode() (map[string]interface{}, error) {
 		return nil, nil
 	}
 
-	// expecting 'sudo' package is installed and /etc/sudoers.d/cagent-dmidecode is present
-	const dmidecodeCmd = "sudo dmidecode"
-	cmd := exec.Command("/bin/sh", "-c", dmidecodeCmd)
+	var dmidecodeCmd []string
+
+	// run dmidecode as command argument to shell
+	dmidecodeCmd = append(dmidecodeCmd, "-c")
+
+	if runtime.GOOS != "darwin" {
```

Comment:
I actually would prefer splitting implementation in compile-time where possible. But we probably should have some agreement with other developers for such cases.

I think, splitting implementation in compile time has a lot of advantages. We already have separate files for these cases (_darwin and _nonwindows), so we can create a function like `executeDmidecode` or `prepareDmidecodeCmd`

Reply:
IMO either case is ok. There was some discussion for that case some while ago and I remember we ended up with this approach. I might be wrong though...

Reply:
I was thinking about the same thing.
I personally also prefer the compile-time way.
The function(s) could be written in a more abstract way and call a function like `smartctlCommand()` which returns  the string for the current OS. Only this helper would sit in `_linux _windows etc.` file

@xlab Any opinion?

Reply:
Made it to detect at compile time